### PR TITLE
Google maps Geocoding typing fix

### PR DESCRIPTION
**Pull request by Cyperghost**

### CODE REVIEW

File: ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts [hunk 1]
```diff
@@ -116,7 +116,7 @@ class Geocoding {
     this.#element.dataset.googleMapsLng = lng.toString();
 
     if (this.#element.hasAttribute("data-google-maps-geocoding-store") && this.#element.form) {
-      this.#store(this.#element.dataset.googleMapsGeocdingStore || "", lat, lng);
+      this.#store(this.#element.dataset.googleMapsGeocodingStore || "", lat, lng);
     }
```
Comment (TimWolla) [inline ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts:120]: This whole logic looks a little odd anyway. Can you check whether the following works:
```
if ("googleMapsGeocodingStore" in this.#element.dataset && this.#element.form) {
  this.#store(this.#element.dataset.googleMapsGeocodingStore, lat, lng);
}
```
This should hopefully cleanly type check.

**Review by TimWolla — CHANGES_REQUESTED** [review]
Please make sure to rebase onto 6.0 and change the target branch to 6.0.

**Comment by TimWolla** [comment]
You'll need to perform an actual rebase, otherwise you're including the changes from master.

Use:

```
git rebase master --onto 6.0
# force push
```

That should do the right thing.

File: ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts [hunk 2]
```diff
@@ -115,8 +115,8 @@ class Geocoding {
     this.#element.dataset.googleMapsLat = lat.toString();
     this.#element.dataset.googleMapsLng = lng.toString();
 
-    if (this.#element.hasAttribute("data-google-maps-geocoding-store") && this.#element.form) {
-      this.#store(this.#element.dataset.googleMapsGeocdingStore || "", lat, lng);
+    if ("googleMapsGeocodingStore" in this.#element.dataset && this.#element.form) {
+      this.#store(this.#element.dataset.googleMapsGeocodingStore!, lat, lng);
```
Comment (TimWolla) [inline ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts:119]: Is the exclamation mark actually necessary here? I hoped that TypeScript is able to narrow down the type.

Comment (Cyperghost) [reply ts/WoltLabSuite/Core/Component/GoogleMaps/Geocoding.ts:119]: My IDE flags this as a possible error, because the value could be undefined.

**Pull request closed without merging**